src/app-nav-bar/types.js

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright (c) 2018-2020 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import * as React from 'react';
+
+// eslint-disable-next-line flowtype/no-weak-types
+type ItemT = any;
+
+export type MainNavItemT = {|
+  active?: boolean,
+  icon?: React.AbstractComponent<{||}>,
+  item: ItemT,
+  mapItemToNode: ItemT => React.Node,
```

Comment:
My thinking is that `mapItemToNode` should be always used to render an item so it has to be required otherwise we'll need to make assumptions about `item` object structure, or use `mapItemToNode` as a backup. The `mapItemToString` is really needed for a11y aria values and a case like `Avatar` rendering where the `name` prop only accepts a string.

Reply:
Can this component not have a reasonable default? I think if this is required we will be seeing lots of non-standard looking menu items

Reply:
It's a good point. I'll add usage of the `mapItemToString` as a backup then.